components/core/tests/test-string_utils.cpp

```diff
@@ -22,6 +23,43 @@ TEST_CASE("to_lower", "[to_lower]") {
     REQUIRE(str == "test123test");
 }
 
+TEST_CASE("replace_unescaped_char", "[replace_unescaped_char]") {
+    auto check = [](std::string in, std::string const& expected) {
+        replace_unescaped_char('\\', '?', '*', in);
+        REQUIRE(in == expected);
+    };
+
+    // initial
+    check(R"(no change)", R"(no change)");
```

Comment:
I feel we can put this under // additional edge cases

let's also replace `backslashes` -> `escapes` (since by default, escape is backslash)

Maybe also add a simple test case when using a custom escape character? you can add it as a new section. (and put existing tests under a section as well)